test/unit/authenticable_getter_name_test.rb

```diff
@@ -0,0 +1,23 @@
+class AuthenticableGetterNameTest < ActiveSupport::TestCase
+  include Knock::Authenticable
+
+  test "check simple model names" do
```

Comment:
Thanks for the unity tests 👍 

I think the tests you added here are covering well the possible scenarios. The only thing I want to add is that testing private methods isn't generally recommended, because the good practices says that it's better to test the behaviors than it's hard implementations. So, I'd suggest either: moving this method to a class which is responsilble only for generating this getter_name (and this class can be isolated in the scope of the class which is using it) OR add these tests to the `authenticate_for` level, so instead asserting if getter_name is the expected, you could assert if `authenticate_for` called/received the correct method... 

Does it make sense for you?

Reply:
Yes, I agree, moved to a separate class 👍🏻